helper/src/components/deployTab.js

```diff
@@ -32,6 +32,7 @@ export default function DeployTab({ defaults, updateFn, tabValues, invalidArray,
     ...(cluster.AksPaidSkuForSLA !== defaults.cluster.AksPaidSkuForSLA && { AksPaidSkuForSLA: cluster.AksPaidSkuForSLA }),
     ...(cluster.SystemPoolType === 'none' ? { JustUseSystemPool: true } : cluster.SystemPoolType !== defaults.cluster.SystemPoolType && { SystemPoolType: cluster.SystemPoolType }),
     ...(cluster.vmSize !== defaults.cluster.vmSize && { agentVMSize: cluster.vmSize }),
+    ...(((cluster.nodepoolName.toLowerCase() !== defaults.cluster.nodepoolName  && cluster.SystemPoolType !== 'none') || ( cluster.SystemPoolType === 'none' && cluster.nodepoolName.toLowerCase() !== defaults.cluster.systemNodepoolName )) && { nodePoolName: cluster.nodepoolName }),
```

Comment:
I think this logic could be simplified now that the name can be used for the system node.

Current logic leads to including the nodepool param when justsystempool is selected.

![image](https://user-images.githubusercontent.com/17914476/202198508-287b0230-2605-4848-badb-dc41f148e2b6.png)

Maybe this works?

```javascript
...(cluster.nodepoolName.toLowerCase() !== defaults.cluster.nodepoolName && { nodePoolName: cluster.nodepoolName })
```

Reply:
This is now handled indeed in same line you referred to. User doesn't have to provide any value if JustUseSysmPool and did not supply a custom name